src/Gui/GLPainter.h

```diff
@@ -24,24 +24,18 @@
 #ifndef GUI_GLPAINTER_H
 #define GUI_GLPAINTER_H
 
-#ifdef FC_OS_WIN32
-#ifndef NOMINMAX
-#define NOMINMAX
-#endif
-#include <windows.h>
-#endif
-#ifdef FC_OS_MACOSX
-#include <OpenGL/gl.h>
-#else
-#include <GL/gl.h>
-#endif
-
 #include <Base/BaseClass.h>
 #include <FCGlobal.h>
-#include <QtOpenGL.h>
 #include <QPoint>
 
 class QPaintDevice;
+class QOpenGLWidget;
+
+using GLenum = unsigned int;
+using GLint = int;
+using GLushort = unsigned short;
+using GLfloat = float;
+using GLdouble = double;
```

Comment:
Can you explain why you want to manually set these types instead of including `OpenGL/gl.h` and `GL/gl.h` which defines them?

Reply:
> Another option might be to have a new header that just forward declares some basic OpenGL types.

If you don't do this, at least include a comment that says why we've taken this (somewhat unorthodox) route.